tox.ini

```
@@ -12,8 +12,8 @@ pytest_cov_args = --cov --cov-report html --cov-report term --cov-report xml
 
 [testenv]
 description = Run tests with {basepython}
-deps = ansible-core
-       -r {toxinidir}/test/requirements.txt
+deps =
+    -r {toxinidir}/test/requirements.txt
```

Try this:
```suggestion
    !devel: ansible-core
    devel: ansible-core @ https://github.com/ansible/ansible/archive/devel.tar.gz
    -r {toxinidir}/test/requirements.txt
```

You shouldn't need to  have this much copied env definitions. Using factors it's more concise.

Those docs got lost in v4 refactoring but it still works: https://github.com/tox-dev/tox/issues/3276.

---

That's an interesting idea, though I generally dislike using factors because if you're not familiar them, they're a bit difficult to understand and aren't as obvious as explicitly defined test envs. I might try this though, only slightly altered (don't want `-devel` available for every test env).